tests/core/full_node/test_tx_processing_queue.py

```diff
@@ -0,0 +1,240 @@
+
+log = logging.getLogger(__name__)
+
+
+# this code generates random spend bundles.
+
+
+def make_standard_spend_bundle(count: int) -> SpendBundle:
+    puzzle_dict: Dict[bytes32, Program] = {}
+    starting_ph: bytes32 = puzzle_hash_for_index(0, puzzle_dict)
+    solution: Program = solution_for_conditions(
+        Program.to([[ConditionOpcode.CREATE_COIN, puzzle_hash_for_index(1, puzzle_dict), uint64(100000)]])
+    )
+    coins = [Coin(bytes32(index.to_bytes(32, "big")), starting_ph, uint64(100000)) for index in range(count)]
+    coin_spends = [CoinSpend(coin, puzzle_dict[starting_ph], solution) for coin in coins]
+    spend_bundle = SpendBundle(coin_spends, blspy.G2Element())
+    return spend_bundle
+
+
+# we only want to call this once as it can take minutes if we call it many times.
+standard_spend_bundle = make_standard_spend_bundle(3)
+
+
+async def get_dummy_peers(
+    server: ChiaServer, num_peers: int = 1, self_hostname: str = "localhost"
+) -> List[WSChiaConnection]:
+    server_port = server.get_port()
+    url = f"wss://{self_hostname}:{server_port}/ws"
+    timeout = aiohttp.ClientTimeout(total=10)
+    session = aiohttp.ClientSession(timeout=timeout)
+    incoming_queue: asyncio.Queue[Tuple[Message, WSChiaConnection]] = asyncio.Queue()
+    # this is all only used to get a web socket that we never use again.
+    config = load_config(server.root_path, "config.yaml")
+    chia_ca_crt_path, chia_ca_key_path = chia_ssl_ca_paths(server.root_path, config)
+    dummy_crt_path = server.root_path / "dummy.crt"
+    dummy_key_path = server.root_path / "dummy.key"
+    generate_ca_signed_cert(
+        chia_ca_crt_path.read_bytes(), chia_ca_key_path.read_bytes(), dummy_crt_path, dummy_key_path
+    )
+    ssl_context = ssl_context_for_client(chia_ca_crt_path, chia_ca_key_path, dummy_crt_path, dummy_key_path)
+    ws = await session.ws_connect(url, autoclose=True, autoping=True, ssl=ssl_context)
+    # we just reuse everything above, except for the peer id.
+    peer_list: List[WSChiaConnection] = []
+    for i in range(num_peers):
+        peer_id = bytes32(token_bytes(32))  # random peer id for each peer.
+        peer_list.append(
+            WSChiaConnection(
+                NodeType.FULL_NODE,
+                ws,
+                server_port,
+                log,
+                True,
+                False,
+                self_hostname,
+                incoming_queue,
+                lambda x, y: None,
+                peer_id,
+                100,
+                30,
+                local_capabilities_for_handshake=capabilities,
+            )
+        )
+    return peer_list
+
+
+def get_transaction_queue_entry(peer: Optional[WSChiaConnection] = None) -> TransactionQueueEntry:
+    sb: SpendBundle = standard_spend_bundle
+    return TransactionQueueEntry(
+        sb,
+        None,
+        sb.name(),
+        peer,
+        False,
+    )
+
+
+class TestTransactionQueue:
```

Comment:
> I think you should remove this class and make all the functions free functions. that way you'll save one level of indentation too

we always use classes when using tests, so i think we should keep it.

Reply:
it's used in many places, most likely because of some original misunderstandings followed by copy+paste. We shouldn't keep bad habits just because they're habits.

Reply:
> > I think you should remove this class and make all the functions free functions. that way you'll save one level of indentation too
> 
> we always use classes when using tests, so i think we should keep it.

unless there is a specific reason not to?

Reply:
in my mind, the main reasons not to are:
1. save one level of indentation
2. save lines of code
3. make it clear to the reader that the tests do not in fact share state (which could be a legitimate reason to use a class)

Reply:
ok that makes sense, will do!